File: typescript/base-hmi-types/src/screens/shapes/HmiText.ts
import { HmiMultilingualText } from "../../common/HmiMultilingualText.js";
import { HmiFont, HmiHorizontalAlignment, HmiProperty, HmiVerticalAlignment } from "../base.js";
import { HmiSurfaceShapeBase } from "./HmiSurfaceShapeBase.js";
import { HmiObjectType } from "../base/HmiObjectType.js";

export class HmiText extends HmiSurfaceShapeBase {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiText;
  }

  text?: HmiProperty<HmiMultilingualText>;
  alternateText?: HmiProperty<HmiMultilingualText>;
  textWrapping?: HmiProperty<number>;
  textTrimming?: HmiProperty<number>;
  font?: HmiFont;
  horizontalAlignment?: HmiProperty<HmiHorizontalAlignment>;
  verticalAlignment?: HmiProperty<HmiVerticalAlignment>;
}

File: src/BaseHmiTypes/Screens/Shapes/HmiText.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Shapes;

public class HmiText : HmiSurfaceShapeBase
{
    public HmiText()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiText;
    }

    public HmiProperty<string>? Text { get; set; }

    public HmiProperty<string>? AlternateText { get; set; }

    public HmiProperty<int>? TextWrapping { get; set; }

    public HmiProperty<int>? TextTrimming { get; set; }

    public HmiFont? Font { get; set; }

    public HmiProperty<HmiHorizontalAlignment>? HorizontalAlignment { get; set; }

    public HmiProperty<HmiVerticalAlignment>? VerticalAlignment { get; set; }
}
